tools/ci/tests/test_pr_preview.py

```diff
@@ -252,75 +252,55 @@ def synchronize(expected_traffic, refs={}):
 
             remote_refs[ref] = revision
 
-    return child.returncode, server.actual_traffic, remote_refs
+    return (
+        method_threw,
+        github_server.actual_traffic,
+        remote_refs,
+    )
 
 
-def detect(event, expected_github_traffic, expected_preview_traffic):
-    env = {
-        'DEPLOY_TOKEN': 'c0ffee'
-    }
-    env.update(os.environ)
-    github_server = MockServer((test_host, 0), expected_github_traffic)
+def deploy(pr_num, revision, expected_github_traffic, expected_preview_traffic):
+    os.environ['GITHUB_TOKEN'] = 'c0ffee'
+
+    github_server = MockServer((TEST_HOST, 0), expected_github_traffic)
     github_port = github_server.server_address[1]
-    preview_server = MockServer((test_host, 0), expected_preview_traffic)
+    preview_server = MockServer((TEST_HOST, 0), expected_preview_traffic, reponse_body_is_json=False)
     preview_port = preview_server.server_address[1]
 
-    with temp_repo() as repo, github_server, preview_server:
-        env['GITHUB_EVENT_PATH'] = repo + '/event.json'
-
-        with open(env['GITHUB_EVENT_PATH'], 'w') as handle:
-            handle.write(json.dumps(event))
-
-        child = subprocess.Popen(
-            [
-                sys.executable,
-                subject,
-                '--host',
-                'http://{}:{}'.format(test_host, github_port),
-                '--github-project',
-                'test-org/test-repo',
-                'detect',
-                '--target',
-                'http://{}:{}'.format(test_host, preview_port),
-                '--timeout',
-                '1'
-            ],
-            cwd=repo,
-            env=env
+    method_threw = False
+    with temp_repo(), github_server, preview_server:
```

Comment:
Does this block still need the temporary repo?

Reply:
Good point, I don't believe so. Done.